lib/resource_waiter.js

```diff
@@ -136,6 +136,16 @@ AWS.ResourceWaiter = inherit({
       callback = params; params = undefined;
     }
 
+    if (params && params.$waiter) {
+      if (params.$waiter.delay) {
+        this.config.delay = params.$waiter.delay;
```

Comment:
Just in case someone wants to enter `0`, maybe use a `typeof x === 'number'` check instead.